client/src/components/CourseView.js

```diff
@@ -2,50 +2,95 @@
-  .filter( instance => instance.id != mockCurrentInstance.id );
+const mockInstitution = 'Aalto University';   // REPLACE SOME DAY? currently this info can't be fetched from database
 
 const CourseView = () => {
   let navigate = useNavigate();
-  let { courseCode } = useParams();
-
+  let { courseId } = useParams();
   const { auth } = useAuth();
 
+  const [courseDetails, setCourseDetails] = useState(null);
+  const [currentInstance, setCurrentInstance] = useState(null);
+  const [instances, setInstances] = useState([]);
+
+  const [animation, setAnimation] = useState(false);
+
+  useEffect(() => {
+    instancesService.getInstances(courseId)
+      .then((data) => {
+        const sortedInstances = data.courseInstances.sort((a, b) => sortingServices.sortByDate(a.startDate, b.startDate));
+        setInstances(sortedInstances);
+        setCurrentInstance(sortedInstances[0]);
+      })
+      .catch((e) => console.log(e.message));
+    
+    courseService.getCourse(courseId)
+      .then((data) => {
+        setCourseDetails(data.course);
+      })
+      .catch((e) => console.log(e.message));
+  }, []);
+
+  useEffect(() => {
+    setAnimation(true);
+  }, [currentInstance]);
+
+  const onChangeInstance = (instance) => {
+    if(instance.id !== currentInstance.id) {
+      setAnimation(false);
+      setCurrentInstance(instance);
+    }
+  };
+
   return(
     <Box sx={{ mr: -4, ml: -4 }}>
-      <Box sx={{ display: 'flex', flexWrap: 'wrap', justifyContent: 'space-between', alignItems: 'center', mb: 4 }}>
-        <Typography variant='h3' sx={{ fontWeight: 'light' }}>{courseCode + ' – ' + mockCurrentInstance.courseData.name.en}</Typography>
-        { /* Only admins and teachers are allowed to create a new instance */
-          (auth.role == 'SYSADMIN' || auth.role == 'TEACHER') && 
-          <Button size='large' variant='contained' onClick={() => { navigate('/fetch-instances/' + courseCode); }}>
-            New instance
-          </Button>
-        }
-      </Box>
-      <Box sx={{ display: 'flex', justifyContent: 'space-evenly', gap: 3 }}>
-        <OngoingInstanceInfo info={mockCurrentInstance} />
-        { /* a different assignment component will be created for students */
-          (auth.role == 'SYSADMIN' || auth.role == 'TEACHER') && 
-          <Assignments assignments={mockAssignmentsClient} formula={'Weighted Average'} instance={mockCurrentInstance} /> /* TODO: Retrieve real formula */
-        }
-      </Box>
-      <Typography variant='h4' align='left' sx={{ fontWeight: 'light', mt: 8, mb: 3 }}>Past Instances</Typography>
-      <InstancesTable data={mockPastInstances} />
+      {courseDetails && currentInstance && instances &&
+        <> 
+          <Box sx={{ display: 'flex', flexWrap: 'wrap', justifyContent: 'space-between', alignItems: 'center', mb: 4 }}>
+            <Typography variant='h3' sx={{ fontWeight: 'light' }}>{courseDetails.courseCode + ' – ' + courseDetails.name.en}</Typography>
```

Comment:
We could look into how to display titles better, since with this format the longer titles look really weird with my screen width. As an example a title looks like this (Sci is on its own line): 

```
MS-A0102 – Differential and Integral Calculus 1
                   (SCI)
```

As a quick fix, in my opinion lining the heading to the left (`textAlign: 'left', py: 1`) would be better but if you don't think that, this doesn't need to be changed.

Reply:
I actually noticed the same thing and agreed! In my infinite wisdom I made the changes to the other branch, so they will come to main when it is merged.